File: scripts/check/check_session.js
import { TmuxInterface } from '../../src/tmux_interface.js';

const SESSION_NAME = 'claude_exec_1749369856326';

async function main() {
    const tmux = new TmuxInterface();
    
    console.log(`Checking for session: ${SESSION_NAME}`);

    try {
        // Check if session exists using canonical TmuxInterface
        const sessionExists = await tmux.sessionExists(SESSION_NAME);

        if (sessionExists) {
            console.log(`✓ Session ${SESSION_NAME} exists`);
            
            // Send Enter key using canonical method
            console.log('Sending Enter key...');
            try {
                const target = tmux.getPaneTarget(SESSION_NAME);
                await tmux.sendKeys(target, '', true); // Send Enter
                console.log(`✓ Successfully sent Enter key to session ${SESSION_NAME}`);
            } catch (error) {
                console.log(`✗ Failed to send Enter key: ${error.message}`);
            }
        } else {
            console.log(`✗ Session ${SESSION_NAME} does not exist`);
        }

        // List all tmux sessions using canonical method
        console.log('\nAll tmux sessions:');
        const sessions = await tmux.listSessions();
        
        if (sessions.length > 0) {
            sessions.forEach(session => {
                console.log(`${session.name}: ${session.windows} windows (${session.attached ? 'attached' : 'detached'})`);
            });
            
            // Look for claude_exec sessions with long timestamps
            console.log('\nClaude exec sessions with long timestamps:');
            sessions.forEach(session => {
                if (session.name.includes('claude_exec_')) {
                    const match = session.name.match(/claude_exec_(\d+)/);
                    if (match && match[1].length > 10) {
                        console.log(`  - ${session.name}: ${session.windows} windows (${session.attached ? 'attached' : 'detached'})`);
                    }
                }
            });
        } else {
            console.log('No tmux sessions found');
        }

        // Also check for any sessions starting with 'claude_'
        console.log('\nAll Claude-related sessions:');
        const claudeSessions = sessions.filter(session => session.name.startsWith('claude_'));
        if (claudeSessions.length > 0) {
            claudeSessions.forEach(session => {
                console.log(`${session.name}: ${session.windows} windows (${session.attached ? 'attached' : 'detached'})`);
            });
        } else {
            console.log('No Claude sessions found');
        }
    } catch (error) {
        console.error('❌ Error:', error.message);
    }
}

// Run the script
main();

File: scripts/check/check_session.py
#!/usr/bin/env python3
import subprocess
import sys

SESSION_NAME = "claude_exec_1749369856326"

def run_command(cmd):
    """Run a command and return output, handling errors gracefully"""
    try:
        result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
        return result.returncode, result.stdout, result.stderr
    except Exception as e:
        return -1, "", str(e)

# Check if session exists
print(f"Checking for session: {SESSION_NAME}")
exit_code, _, _ = run_command(f"tmux has-session -t {SESSION_NAME}")

if exit_code == 0:
    print(f"✓ Session {SESSION_NAME} exists")
    
    # Send Enter key
    print("Sending Enter key...")
    exit_code, stdout, stderr = run_command(f"tmux send-keys -t {SESSION_NAME} Enter")
    
    if exit_code == 0:
        print(f"✓ Successfully sent Enter key to session {SESSION_NAME}")
    else:
        print(f"✗ Failed to send Enter key: {stderr}")
else:
    print(f"✗ Session {SESSION_NAME} does not exist")

# List all tmux sessions
print("\nAll tmux sessions:")
exit_code, stdout, stderr = run_command("tmux list-sessions")

if exit_code == 0 and stdout:
    print(stdout)
    
    # Look for claude_exec sessions with long timestamps
    print("\nClaude exec sessions with long timestamps:")
    for line in stdout.split('\n'):
        if 'claude_exec_' in line and len(line.split('claude_exec_')[1].split(':')[0]) > 10:
            print(f"  - {line}")
else:
    print("No tmux sessions found")

# Also check for any sessions starting with 'claude_'
print("\nAll Claude-related sessions:")
exit_code, stdout, stderr = run_command("tmux list-sessions | grep claude_")
if exit_code == 0 and stdout:
    print(stdout)
else:
    print("No Claude sessions found")